### PR TITLE
System modal system api

### PR DESCRIPTION
**Pull request by trautamaki**

### CODE REVIEW

File: app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt [hunk 1]
```diff
@@ -234,7 +234,9 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
     override fun onDestroy() {
         super.onDestroy()
 
-        notificationViewModel.unbindService(this)
+        if (NotificationUtils.notificationPermissionGranted(this)) {
```
Comment (trautamaki) [inline app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:243]: @luca020400 should this be done better? What if the user disables notification access? I noticed one crash with unbinding with it not being bind in the first place.

Comment (luca020400) [reply app/src/main/java/org/lineageos/tv/launcher/MainActivity.kt:243]: Ah that's on me yeah, maybe there's a smarter way to do it but simply checking if you have the perm before unbinding will do

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -84,6 +84,18 @@
             </intent-filter>
         </service>
 
+        <service
```
Comment (luca020400) [inline app/src/main/AndroidManifest.xml]: Not sure how I feel about it, a11y services are sad.
I know you want to preserve the non-system app compat but if we're targeting Lineage I wouldn't want it.

**Comment by luca020400** [comment]
I think ADB might be a nice addition, but not really related to this PR :)
Looks good now

**Review by luca020400 — APPROVED** [review]

**Comment by trautamaki** [comment]
> I think ADB might be a nice addition, but not really related to this PR :) Looks good now

Yep, I will do some work on the platform side and add that later here.

**Pull request merged**